src/data/polynomial/degree/definitions.lean

```diff
@@ -1156,6 +1156,24 @@ else if hq0 : q = 0 then  by simp only [hq0, degree_zero, mul_zero, with_bot.add
 else degree_mul' $ mul_ne_zero (mt leading_coeff_eq_zero.1 hp0)
     (mt leading_coeff_eq_zero.1 hq0)
 
+/-- degree is "order preserving" from `(R[X], ∣)` to `(with_bot ℕ, ≤)` if q is non-zero -/
+lemma degree_le_of_dvd (hqn0 : q ≠ 0) (hpq : p ∣ q) : degree p ≤ degree q :=
```

Comment:
Is there some way to prove `nat_degree_le_of_dvd` in terms of this? Or perhaps to use the proof at `nat_degree_le_of_dvd` to inform a shorter proof here?

I think this lemma probably belongs next to `nat_degree_le_of_dvd` either way.